File: server/src/main/java/gg/mineral/server/network/packet/play/serverbound/PlayerLookPacket.java
package gg.mineral.server.network.packet.play.serverbound;

import gg.mineral.api.network.connection.Connection;
import gg.mineral.api.network.packet.Packet;
import io.netty.buffer.ByteBuf;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.val;
import lombok.experimental.Accessors;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Accessors(fluent = true)
public final class PlayerLookPacket implements Packet.INCOMING {
    private float yaw, pitch;
    private boolean onGround;

    @Override
    public void received(Connection connection) {
        val player = connection.getPlayer();

        if (player == null)
            return;

        player.setYaw(yaw);
        player.setPitch(pitch);
        player.setOnGround(onGround);
    }

    @Override
    public void deserialize(ByteBuf is) {
        yaw = is.readFloat();
        pitch = is.readFloat();
        onGround = is.readBoolean();
    }
}

File: server/src/main/kotlin/gg/mineral/server/network/packet/play/serverbound/PlayerLookPacket.kt
package gg.mineral.server.network.packet.play.serverbound

import gg.mineral.api.network.connection.Connection
import io.netty.buffer.ByteBuf

class PlayerLookPacket(var yaw: Float = 0f, var pitch: Float = 0f, onGround: Boolean = false) : PlayerPacket(onGround) {

    override fun receivedSync(connection: Connection) {
        connection.player?.let {
            it.yaw = yaw
            it.pitch = pitch
        }
        super.receivedSync(connection)
    }

    override fun deserialize(`is`: ByteBuf) {
        yaw = `is`.readFloat()
        pitch = `is`.readFloat()
        super.deserialize(`is`)
    }
}
